routing/router.go

```diff
@@ -1642,7 +1628,10 @@ func (r *ChannelRouter) SendPayment(payment *LightningPayment) ([32]byte,
 
 	// Since this is the first time this payment is being made, we pass nil
 	// for the existing attempt.
-	return r.sendPayment(nil, payment, paySession)
+	return r.sendPayment(
+		payment.Amount, payment.FeeLimit, payment.PaymentHash,
+		payment.PayAttemptTimeout, paySession,
```

Comment:
It feels redundant to pass in the same parameters that are already contained in `paySession`, but probably good enough for now